src/ProjectTemplates/Web.ProjectTemplates/Components-CSharp.csproj.in

```diff
@@ -0,0 +1,35 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <!--#if UseClient -->
+    <TargetFrameworks>${DefaultNetCoreTargetFramework};${DefaultNetCoreTargetFramework}-browser</TargetFrameworks>
+    <!--#else -->
+    <TargetFramework>${DefaultNetCoreTargetFramework}</TargetFramework>
+    <!--#endif -->
+    <Nullable>enable</Nullable>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <NoDefaultLaunchSettingsFile Condition="'$(ExcludeLaunchSettings)' == 'True'">True</NoDefaultLaunchSettingsFile>
+    <RootNamespace Condition="'$(name)' != '$(name{-VALUE-FORMS-}safe_namespace)'">ComponentsWebAssembly-CSharp.Server</RootNamespace>
+    <AssemblyName Condition="'$(name)' != '$(name{-VALUE-FORMS-}safe_namespace)'">`$(AssemblyName.Replace(' ', '_'))</AssemblyName>
+    <!--#if UseClient && PWA -->
+    <ServiceWorkerAssetsManifest>service-worker-assets.js</ServiceWorkerAssetsManifest>
+    <!--#endif -->
+  </PropertyGroup>
+  <!--#if UseClient -->
+
+  <ItemGroup Condition="'$([MSBuild]::GetTargetPlatformIdentifier($(TargetFramework)))' == 'browser'">
+    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="${MicrosoftAspNetCoreComponentsWebAssemblyVersion}" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$([MSBuild]::GetTargetPlatformIdentifier($(TargetFramework)))' == ''">
+    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.Server" Version="${MicrosoftAspNetCoreComponentsWebAssemblyServerVersion}" />
+  </ItemGroup>
+  <!--#endif -->
+  <!--#if UseClient && PWA -->
+
```

Comment:
```suggestion

    <!--#if UseClient && PWA -->
```

Just to make the spacing even in both cases.

Reply:
GitHub is showing this diff weirdly. I intended to swap lines 28 and 29, so the leading blank line is outside the "if" block.

Reply:
You want the empty line inside the if block. Otherwise, if you don't, you end up with multiple empty lines in the templates. the pattern in the templates is to put the required spacing inside the conditional blocks.

Reply:
To my best understanding, this is mistaken. If `UseClient`==true and `PWA`=false, it looks as if this template would render as:

```xml
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFrameworks>${DefaultNetCoreTargetFramework};${DefaultNetCoreTargetFramework}-browser</TargetFrameworks>
    <Nullable>enable</Nullable>
    <ImplicitUsings>enable</ImplicitUsings>
    <NoDefaultLaunchSettingsFile Condition="'$(ExcludeLaunchSettings)' == 'True'">True</NoDefaultLaunchSettingsFile>
    <RootNamespace Condition="'$(name)' != '$(name{-VALUE-FORMS-}safe_namespace)'">ComponentsWebAssembly-CSharp.Server</RootNamespace>
    <AssemblyName Condition="'$(name)' != '$(name{-VALUE-FORMS-}safe_namespace)'">`$(AssemblyName.Replace(' ', '_'))</AssemblyName>
  </PropertyGroup>

  <ItemGroup Condition="'$([MSBuild]::GetTargetPlatformIdentifier($(TargetFramework)))' == 'browser'">
    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="${MicrosoftAspNetCoreComponentsWebAssemblyVersion}" />
  </ItemGroup>

  <ItemGroup Condition="'$([MSBuild]::GetTargetPlatformIdentifier($(TargetFramework)))' == ''">
    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.Server" Version="${MicrosoftAspNetCoreComponentsWebAssemblyServerVersion}" />
  </ItemGroup>
</Project>
```

Notice there is a missing blank line before `</Project>` (it is not balanced with the blank line after the opening `<Project>`), whereas that problem does not occur if `PWA`==true.

Reply:
However my suggestion was also incomplete - just swapping 28 and 29 doesn't fix this. More blank-line changes are needed. Each of the top-level `if` blocks need to have the same blank-line pattern, i.e.,:

 * Either all the top-level ones start with a blank line and *don't* end with one
 * Or all the top-level ones end with a blank line and *don't* start with one